### PR TITLE
Added ch3 minimum exercise

### PR DESCRIPTION
**Pull request by clkwrkunvrs**

### CODE REVIEW

File: ch3/minimum.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import assert from 'assert';
+const minimum = (a,b) => {
```
Comment (CelestineKao) [inline ch3/minimum.js]: add a newline between these two! there should always be a newline after all the imports.

File: ch3/minimum.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  //return b ^ ((a ^ b) & -(a < b)); //bitwise min
```
Comment (CelestineKao) [inline ch3/minimum.js]: delete

File: ch3/beamCounting.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  }, 0);
+}
+
+/*const countChar = (inputString, charToFind) => {
+  let numChar = -1;
+  inputString.split('').forEach(character => {
+    character === charToFind ? numChar++ : null;
+  });
+  return numChar;
+}*/
```
Comment (CelestineKao) [inline ch3/beamCounting.js]: delete

File: ch3/minimum.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  return a <= b ? a : b;
+}
+
+assert(minimum(1,2) == 1);
+assert(minimum(2,1) == 1);
+assert(minimum(-1,1) == -1);
+assert(minimum(2,2) == 2);
+assert(minimum(0,0) == 0);
+assert(minimum(-1,-1) == -1);
```
Comment (CelestineKao) [inline ch3/minimum.js]: always use `===` instead of `==` when comparing values. `==` does type conversion if necessary which can result in some unexpected behaviors, so always just use `===`. see this [SO discussion](https://stackoverflow.com/questions/359494/which-equals-operator-vs-should-be-used-in-javascript-comparisons) for some of the weird cases.

Comment (CelestineKao) [reply ch3/minimum.js]: love the use of `assert` though!

File: ch3/recursion.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+const assert =  require('assert');
```
Comment (CelestineKao) [inline ch3/recursion.js]: out of curiosity, why did you change to a `require` when you used `import` in `minimum.js`? 

i'd suggest always using `import` unless there's some strong reason that makes `require` necessary. see this [SO discussion](https://stackoverflow.com/questions/31354559/using-node-js-require-vs-es6-import-export) on why `import` is better.

File: ch3/recursion.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  /*return (n === 0) ? true :
+  (n === 1) ? false :
+  isEven(n - 2);*/
+}
+
+//Is there harm in using assrt or jest in this case?
+//I can see how it's a time suck but if you have
+//many function calls, you can get lost on which one is failing
+//and since it's a recursive call, the input value changes from your original
+//input each time
```
Comment (CelestineKao) [inline ch3/recursion.js]: no harm, candidates normally use `assert` instead of `jest` since `jest` requires a bit more scaffolding. if you find yourself in an interview where you can't import `'assert'` for whatever reason, you can also do something like:
```js
[
 [2, true],
 [1, false],
 [-1, false]
].forEach(([input, expectedOutput]) => `For input ${input}, expected ${expectedOutput}, received ${isEven(input)}`);
```

File: ch3/recursion.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
+const assert =  require('assert');
+/**
+ * checks if input value is even
+ * @param {*} n - input value
+ * @returns true if even, false if not
+ */
```
Comment (CelestineKao) [inline ch3/recursion.js:7]: i like docstrings, but i wouldn't worry about them in the context of an interview since they just take up time! but yes, definitely +1 to docstrings when working.

File: ch3/recursion.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+const isEven = (n) => {
+  n = Math.abs(n);
+
+  //these are so short. Can i just keep them one line? Would anyone really care?
```
Comment (CelestineKao) [inline ch3/recursion.js]: yes one line is totally fine in these cases

File: ch3/recursion.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  if (n === 0) return true; 
+  else if (n === 1) return false;
+  return isEven(n-2);
+  //which one? first one easier to read
+  /*return (n === 0) ? true :
+  (n === 1) ? false :
+  isEven(n - 2);*/
```
Comment (CelestineKao) [inline ch3/recursion.js]: always bias towards the easier one to read! the easier it is to read, the easier it is for anyone to maintain. there are some scenarios where making code _fast_ means very unreadable code, but don't worry about those scenarios right now.

File: ch3/recursion.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+ * @param {*} n - input value
+ * @returns true if even, false if not
+ */
+const isEven = (n) => {
+  n = Math.abs(n);
+
+  //these are so short. Can i just keep them one line? Would anyone really care?
+  if (n === 0) return true; 
+  else if (n === 1) return false;
+  return isEven(n-2);
```
Comment (CelestineKao) [inline ch3/recursion.js]:
```suggestion
  return isEven(n - 2);
```

always add 1 space on both sides of operators.

**Review by CelestineKao — CHANGES_REQUESTED** [review]

File: ch3/minimum.js (hunk 1 above)
Comment (clkwrkunvrs) [reply ch3/minimum.js]: Ok

File: ch3/minimum.js (hunk 4 above)
Comment (clkwrkunvrs) [reply ch3/minimum.js]: You got me. I forgot it here.

File: ch3/recursion.js (hunk 5 above)
Comment (clkwrkunvrs) [reply ch3/recursion.js]: Because my linter in vscode makes colors for suggestions and I want them to make go away. This one says you can change the file to an ES module and change the syntax so I just accept the quick fix.

File: ch3/recursion.js (hunk 6 above)
Comment (clkwrkunvrs) [reply ch3/recursion.js]: Do I need to chain this with assert? Or it prints on its own?

File: ch3/recursion.js (hunk 9 above)
Comment (clkwrkunvrs) [reply ch3/recursion.js]: Ok

File: ch3/minimum.js (hunk 4 above)
Comment (clkwrkunvrs) [reply ch3/minimum.js]: Thank you for that SO post. That's very helpful

File: ch3/recursion.js (hunk 5 above)
Comment (clkwrkunvrs) [reply ch3/recursion.js]: i changed it back and read your SO discussion. Thank you

File: ch3/recursion.js (hunk 6 above)
Comment (clkwrkunvrs) [reply ch3/recursion.js]: oh i see. I was confused because there was no console.log in this. I got it now.
